Sources/System/FileHelpers.swift

```diff
@@ -79,6 +79,75 @@ extension FileDescriptor {
     }
   }
 
+  /// Reads bytes at the current file offset into a buffer until the buffer is filled.
```

Comment:
What happens if the file runs out of bytes?

Reply:
I've used libraries where the fill-this-buffer-by-reading-bytes operation would signal an error if it runs out of bytes and I used ones where it would return a partial buffer. Generally I found the erroring variant more convenient, as in the other case I'd usually just end up having to signal an error myself. (This is supposing that the regular partial `read` is available as a separate operation, like is the case with System.)  So from a usability standpoint I'd prefer this threw an error on EOF. 

Unfortunately, there is no errno value that represents the EOF condition -- the original i/o interfaces do not treat this as an error. So we'll have to define a custom `Error` type if we want to do this.

(We'll need to step carefully here, as I'd prefer if we did not start a slippery slope where we end up with a full blown File abstraction within System -- this library ought to simply sanitize system interfaces, not reinvent them.)

Reply:
I think returning the actual number of bytes read is better than throwing. How do FIFOs, etc. work in this respect?

> Upon successful completion, read(), readv(), pread(), and preadv() return
     the number of bytes actually read and placed in the buffer.  The system
     guarantees to read the number of bytes requested if the descriptor
     references a normal file that has that many bytes left before the end-of-
     file, but in no other case.

>RETURN VALUES
>     If successful, the number of bytes actually read is returned.  Upon reading
     end-of-file, zero is returned.  Otherwise, a -1 is returned and the global
     variable errno is set to indicate the error.

Reply:
The current state of this PR is that it will return an `Int` for the number of bytes read, which will be _at most_ `buffer.count`. If there were enough bytes in the file to fill the buffer then it does fill the buffer and returns `buffer.count`, if EOF is encountered then it will _not_ throw, but return the number of bytes read into the buffer, which will be less than `buffer.count`.

How do we feel about that? LMK if you want something different.